File: IAIso-v5.0/core/iaiso-ruby/lib/iaiso/consent/revocation_list.rb
# frozen_string_literal: true

require "monitor"
require "set"

module IAIso
  module Consent
    # Thread-safe set of revoked JTIs.
    class RevocationList
      include MonitorMixin

      def initialize
        super
        @set = Set.new
      end

      def revoke(jti)
        synchronize { @set << jti }
      end

      def revoked?(jti)
        synchronize { @set.include?(jti) }
      end

      def clear
        synchronize { @set.clear }
      end
    end
  end
end

File: IAIso-v5.0/core/iaiso-php/src/Consent/RevocationList.php
<?php

declare(strict_types=1);

namespace IAIso\Consent;

/** Thread-safe-ish set of revoked JTIs. */
final class RevocationList
{
    /** @var array<string,bool> */
    private array $set = [];

    public function revoke(string $jti): void
    {
        $this->set[$jti] = true;
    }

    public function isRevoked(string $jti): bool
    {
        return isset($this->set[$jti]);
    }

    public function clear(): void
    {
        $this->set = [];
    }
}
